From 96c7d66f7fac7628067dd9a6ac62589ae571f0fa Mon Sep 17 00:00:00 2001
From: Balamurugan Mahalingam <quic_bmahalin@quicinc.com>
Date: Thu, 13 Oct 2022 17:33:02 -0700
Subject: [PATCH 2/7] ath12k: Update iowrite32 to write_relaxed to boost
 performance

Use writel_relaxed instead iowrite32 to avoid the memory barrier
instuction and slowdown the cpu execution performance.
This helps improve the throughput by 150Mbps

Signed-off-by: Balamurugan Mahalingam <quic_bmahalin@quicinc.com>
---
 drivers/net/wireless/ath/ath12k/hal.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/net/wireless/ath/ath12k/hal.h b/drivers/net/wireless/ath/ath12k/hal.h
index f1525ed..4922ead 100644
--- a/drivers/net/wireless/ath/ath12k/hal.h
+++ b/drivers/net/wireless/ath/ath12k/hal.h
@@ -1211,7 +1211,7 @@ void ath12k_hal_srng_access_umac_src_ring_end_nolock(void *mem,
 
 	srng->u.src_ring.last_tp =
 			*(volatile u32 *)srng->u.src_ring.tp_addr;
-	iowrite32(srng->u.src_ring.hp, mem + HAL_DP_REG_WINDOW_OFFSET
+	writel_relaxed(srng->u.src_ring.hp, mem + HAL_DP_REG_WINDOW_OFFSET
 		  + (offset & WINDOW_RANGE_MASK));
 	srng->timestamp = jiffies;
 }
-- 
2.17.1

